src/ansys/fluent/core/services/datamodel_se.py

```diff
@@ -1514,6 +1531,7 @@ def __iter__(self) -> Iterator[PyMenu]:
             )
 
     def _get_item(self, key: str) -> PyMenu:
+        key = self._check_via_internal_names(key)
```

Comment:
`_get_child_object_display_names()` is being called twice in each of these methods. To deal with this, consolidate the code that handles checking the provided key. You should also consolidate the code that evaluates the `child_path` which is currently duplicated across these methods.

Reply:
@prmukherj If this is complicated, we should through the name at the server and let it raise any exceptions. Clients should be simple (we want more of them).